R/rerun.R

```diff
@@ -15,22 +16,24 @@
 #'   There is one special case: if there's a single unnamed input, the second
 #'   level list will be dropped. In this case, `rerun(n, x)` behaves like
 #'   `replicate(n, x, simplify = FALSE)`.
-#'
-#' @section Lifecycle:
-#'
-#' `rerun()` is in the questioning lifecycle stage because we are no
-#' longer convinced NSE functions are a good fit for purrr. Also,
-#' `rerun(n, x)` can just as easily be expressed as `map(1:n, ~ x)`
-#' (with the added benefit of being passed the current index as
-#' argument to the lambda).
-#'
 #' @export
 #' @examples
+#' # old
 #' 10 %>% rerun(rnorm(5))
+#' # new
+#' 1:10 %>% map(~ rnorm(5))
+#'
+#' # old
 #' 10 %>%
 #'   rerun(x = rnorm(5), y = rnorm(5)) %>%
 #'   map_dbl(~ cor(.x$x, .x$y))
+#' # new
+#' 1:10 %>%
+#'   map(~ list(x = rnorm(5), y = rnorm(5))) %>%
+#'   map_dbl(~ cor(.x$x, .x$y))
 rerun <- function(.n, ...) {
+  lifecycle::deprecate_warn("0.4.0", "rerun()", "map()")
```

Comment:
I'm a bit worried that this function might be used in scripts. Should we just supersede?

Reply:
You only see deprecation warnings every 8 hours so I don't feel like it's super annoying. What about if I just made more prominent advice?

Reply:
ah yes it might be worth a few more details about the switch.